Disabled the dependencies metadata from Singed APK for kiwix app.

MohitMaliFtechiz opened the pull request:
Parent Issue #4053

See https://github.com/kiwix/kiwix-android/issues/4053#issuecomment-2456951610

* We have disabled the dependencies metadata from the signed APK(For Kiwix App) since this APK we are posting on IzzyOnDroid, GitHub, and our website so the metadata is not needed for this. For the app bundle: We have kept the dependencies metadata since it will upload on the Google PlayStore where the dependency metadata is analyzed to provide actionable feedback on potential issues with project dependencies.
* Added the information about permissions used in the application in the README file to show users why and where we are using these permissions.
* Updated the outdated content of the README file(It was showing file picker is not available in the PS variant).

kelson42 reviewed (changes requested):
@MohitMaliFtechiz Sorry for the late review, but I have many remarks:
* Please document - in the code where the permission is requested - clearly why each of these permissions are needed. It's important this know how is clearly documented
* I have no clue what "he dependencies metadata" concretly is and is for. Please tell me.

MohitMaliFtechiz commented:
@kelson42 I have updated the comments in the Code and also updated the PR description as well.

Building dependency metadata for APKs and app bundles is by default enabled in Android. This is basically the metadata of the dependencies used in our project. The playStore analyzes this metadata of dependencies and provides actionable feedback on potential issues or improvements with the project dependencies.

For website, Github, and IzzyOnDroid APKs this metadata is not required since there is no use of this metadata. But on Play Store we are uploading the bundle for the Kiwix app so in the app bundle we need this metadata. So for the app bundle, we have kept this metadata so that play store can give actionable feedbacks on the project's dependencies. https://developer.android.com/build/dependencies#dependency-info-play

kelson42 commented:
@MohitMaliFtechiz I don't see the commenta in the code (at least not in this PR)!

MohitMaliFtechiz commented:
@kelson42 I have added these comments in the `build.gradle` file of the app module(For the kiwix app) https://github.com/kiwix/kiwix-android/pull/4057/files#diff-8cff73265af19c059547b76aca8882cbaa3209291406f52df1dafbbc78e80c46R90 are enough to describe this permission? and where this permission is requested, and why we are disabling this for APK, and why we are enabling this for app bundle.

kelson42 commented:
@MohitMaliDeveloper Your last commit has nothing to do with android permissions! Reread the comment of @IzzySoft please. I don't want do have devs (and maybe even users) keep asking why this kind or this other kind of permission is requested. Please document this clearly so they don't have to ask and can just read the appropriate documentation.

BTW, I first requested to have this documentation in the code. I believe now this is a bad idea because normal users will never find it. Please create a new section in the README dedicated zo this.

MohitMaliFtechiz commented:
> **@MohitMaliDeveloper Your last commit has nothing to do with Android permissions! Reread the comment of @IzzySoft please. I don't want do have devs (and maybe even users) keep asking why this kind or this other kind of permission is requested. Please document this clearly so they don't have to ask and can just read the appropriate documentation.
BTW, I first requested to have this documentation in the code. I believe now this is a bad idea because normal users will never find it. Instead, Please create a new section in the README dedicated zo this.**

Hooo!! I got you now, you are talking about Android permissions (ACCESS_FINE_LOCATION, NEARBY_WIFI_DEVICES, MANAGE_EXTERNAL_STORAGE). 

Since this PR is for removing the dependencies metadata from the APKs, and you had asked to document in the code so I thought you were talking about adding comments for excluding the metadata. Because documenting this permission in code does not have effect on the users.

> I believe now this is a bad idea because normal users will never find it. Instead, Please create a new section in the README dedicated zo this.

Got it.

MohitMaliFtechiz commented:
@kelson42 I have added the information about the permissions used in the application in the README file. Also, updated the outdated content of the README file. As now we have the file picker functionality in our PS variant, and README was showing that there is no file picker in the PS variant.
